node/utils/add_generator_if_missing.ts

```diff
@@ -0,0 +1,6 @@
+import type { Declaration } from '../declaration.js'
+
+const addGeneratorIfMissing = (functions?: Declaration[]) =>
```

Comment:
I'm curious why you decided to move this into its own util file, since it's just extending an object with an additional property?

Reply:
🤔 I am used to splitting things up this way, but now that you mention it, it doesn't make sense in this case no :)